### PR TITLE
completions: mixed bag of updates/additions

### PR DESCRIPTION
**Pull request by ammgws**
Mixed bag of completion updates/additions

### CODE REVIEW

File: share/completions/telnet.fish [hunk 1]
```diff
@@ -1,25 +1,51 @@
...
 complete -c telnet -s l -x -a "(__fish_complete_users)" -d (_ "User login")
+complete -c telnet -s L -d (_ "Specifies an 8-bit data path")
 complete -c telnet -s n -x -d (_ "Log to tracefile")
-complete -c telnet -s x -d (_ "Turn on encryption")
 complete -c telnet -s r -d (_ "User interface similar to rlogin")
-complete -c telnet -s k -x -d (_ "Use Kerberos realm for authentication")
 
+switch (uname -s)
+  case FreeBSD NetBSD Linux
+    complete -c telnet -s d -d (_ "Sets debug mode")
```
Comment (faho) [inline share/completions/telnet.fish]: These completions set a terrible precedent - they not only use `_` explicitly (which you don't have to, we already translate descriptions internally, whereas `_` will call the gettext *command*), but they also don't quote it.

This means if you had a translation that either had no lines or more than one line you'd get a syntax error.

Please just use 

```fish
complete -c telnet -s d -d "Sets debug mode"
```

File: share/completions/whatis.fish [hunk 2]
```diff
@@ -1,2 +1,14 @@
-
 complete -xc whatis -a "(__fish_complete_man)"
+
+complete -c whatis -s d -l debug -d "Debug"
+complete -c whatis -s v -l verbose -d "Verbose"
+complete -c whatis -s r -l regex -d "Interpret each keyword as a regex"
+complete -c whatis -s w -l wildcard -d "The keyword(s) contain wildcards"
+complete -c whatis -s l -l long -d "Do not trim output to terminal width"
+complete -rc whatis -s C -l config-file -d "Configuration file"
+complete -xc whatis -s L -l locale -a "(command -sq locale; and locale -a)" -d "Set locale"
```
Comment (faho) [inline share/completions/whatis.fish]: Good idea with the check!

File: share/completions/whatis.fish [hunk 3]
```diff
@@ -1,2 +1,14 @@
-
 complete -xc whatis -a "(__fish_complete_man)"
+
+complete -c whatis -s d -l debug -d "Debug"
+complete -c whatis -s v -l verbose -d "Verbose"
+complete -c whatis -s r -l regex -d "Interpret each keyword as a regex"
+complete -c whatis -s w -l wildcard -d "The keyword(s) contain wildcards"
```
Comment (faho) [inline share/completions/whatis.fish]: "Allow wildcards"?

File: share/completions/telnet.fish (hunk 1 above)
Comment (ammgws) [reply share/completions/telnet.fish]: Yeah I will remove them. Is there any place for them anymore in completions? There's probably some other older files which still use them.

File: share/completions/whatis.fish (hunk 3 above)
Comment (ammgws) [reply share/completions/whatis.fish]: Thanks. It did seem awkward at the time.

File: share/completions/telnet.fish (hunk 1 above)
Comment (faho) [reply share/completions/telnet.fish]: In descriptions definitely no. Like I said we translate them in the C++. Which is faster and means not having to type `(_` all the time.

I don't think there are many other uses.

**Comment by faho** [comment]
Merged, thanks!

**Pull request merged**